src/client.rs

```diff
@@ -391,6 +391,88 @@ fn need_retry(e: &Error) -> bool {
     }
 }
 
+struct MyStream<'a, T: Presto + 'static> {
+    inner: QueryInternal<'a, T>,
+    phantom: std::marker::PhantomData<T>,
+}
+
+enum QueryInternal<'a, T: Presto + 'static> {
+    Query {
+        sql: String,
+        client: &'a Client,
+    },
+    Next {
+        already: Vec<T>,
+        next: Option<String>,
+        client: &'a Client,
+    },
+}
```

Comment:
Can we merge `MyStream` and `QueryInternal` into something like `QueryStream`.
I can't see why a wrapper is necessary.